config/v1/types_infrastructure.go

```diff
@@ -457,6 +462,32 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+type ClusterDNSConfig struct {
+	// APIServerDNSConfig contains DNS configuration for the API Server.
+	// +optional
+	APIServerDNSConfig []DNSConfig `json:"apiServerDNSConfig,omitempty"`
+
+	// InternalAPIServerDNSConfig contains DNS configuration for the Internal API Server.
+	// +optional
+	InternalAPIServerDNSConfig []DNSConfig `json:"internalAPIServerDNSConfig,omitempty"`
+
+	// IngressDNSConfig contains DNS configuration for cluster services.
+	// +optional
+	IngressDNSConfig []DNSConfig `json:"ingressDNSConfig,omitempty"`
+}
+
+// DNSConfig contains the LB IP and DNS Record type to configure a DNS entry.
+type DNSConfig struct {
+	// recordType is the DNS record type.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[A-Za-z]+$`
+	RecordType string `json:"recordType"`
+	// lBIPAddress is the Load Balancer IP address for DNS config
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Format=ip
+	LBIPAddress string `json:"lbIPAddress"`
```

Comment:
I would avoid the abbreviation here, `loadBalancerIPAddress` would be better IMO, WDYT?

> // +kubebuilder:validation:Format=ip

Have you tested this? I'm not sure this works, you might need a patch for this. Can you add some integration tests that check that these work (see the yaml file for examples, `.testsuite.yaml` for the appropriate resource)